src/luacheck/detect_cyclomatic_complexity.lua

```diff
@@ -0,0 +1,139 @@
+local utils = require "luacheck.utils"
+
+local CyclomaticComplexityMetric = utils.class()
+
+function CyclomaticComplexityMetric:incr_decisions(count)
+    if self.count then
```

Comment:
`self.count` is always set when this is called, it's a bug if it's not. So it's better to remove this condition IMO.

Reply:
ok